### DISCUSSION TITLE
Cannot use `fullwidth` and `stacked` at the same time

### DISCUSSION THREAD

**Discussion by mrleblanc101** [laravel/nova-issues#5440] [Ideas & Feature Requests]
- Laravel Version: 9.52.4
- Nova Version: 4.22.2

### Description:
Hi,
In our app we want to make a custom field, the field should be stacked (label above the field) and fullwidth.
Currently, this is not possible.

Stacked field:
![stacked](https://user-images.githubusercontent.com/17392251/228347115-5780eeb5-4269-4f90-935e-b1a1efd84ecf.png)

Fullwidth:
![fullscreen](https://user-images.githubusercontent.com/17392251/228347146-0367f6f1-ceda-4e33-82a6-a8c5fef22cb7.png)

Stacked and fullwidth:
![stackedfullscreenbox](https://user-images.githubusercontent.com/17392251/228347736-a83b8e51-1a02-4288-94a7-a2373a4896f3.png)

Here is what it should look like:
![whatitshouldbestackedfullscreen](https://user-images.githubusercontent.com/17392251/228347793-8bd019e7-22b8-41be-bb08-d8016a667f2f.png)

### Detailed steps to reproduce the issue on a fresh Nova installation:
```
\Hgm\Media\MultipleMedia::make('Media')->fullWidth()->stacked()->rules('required')->required()
```

**Comment by crynobone** [comment]
### Default

![image](https://user-images.githubusercontent.com/172966/228391585-48ae06ec-1a73-491a-a7d4-cdd431338de7.png)

### Stacked

![image](https://user-images.githubusercontent.com/172966/228391651-e9dd48bc-dd6e-4c42-b9c2-064c6e4de7fc.png)

### Full-width

![image](https://user-images.githubusercontent.com/172966/228391699-3008b4db-b7b1-448b-839b-ab33ec050c78.png)

### Stacked + Full-width

![image](https://user-images.githubusercontent.com/172966/228391764-72ce2791-a597-4028-955c-97a33ab755c4.png)

--------------------------

We don't handle issues generated by 3rd party custom fields, you have 100% control over the UI and should be able to tweak the styling depending on your requirement.

**Comment by mrleblanc101** [comment]
@crynobone This issue is with Nova, not the custom field. You can see the issue on your own screenshot.

Clearly this (fullwidth + stacked) is NOT fullwidth lol
![image](https://user-images.githubusercontent.com/17392251/228392193-dd5075b3-047b-4395-8716-1e3f4b980614.png)

I have no control over the width of the field other than those 2 options.

**Comment by mrleblanc101** [comment]
This is clearly not fullwidth.

![image](https://user-images.githubusercontent.com/17392251/228392472-94ca4cec-ecbd-4252-ae45-a016a015bd3c.jpeg)

My field can't be fullwidth because of how Nova wrap the fields as seen here:
![image](https://user-images.githubusercontent.com/17392251/228392692-43cc0314-fc5e-43b9-9638-6544396166a7.png)

**Comment by mrleblanc101** [comment]
So the label is w-1/5 and the field is w-4/5. 

When using fullwidth and stacked togerther; The flex direction is changed to column, but the field is still w-4/5 instead of w-5/5 or w-full which defeat the purpose of calling if "fullwidth".

**Comment by crynobone** [comment]
This behaviour has been reviewed before and is considered as expected. Create a custom CustomField component if you want a different behaviour.

**Comment by mrleblanc101** [comment]
@crynobone Kind of a bad API name don't you think ? Calling something fullwidth when it's clearly 4/5.

**Comment by mrleblanc101** [comment]
If you tell me this is so all field are aligned. Why is fullwidth when not stacked make the field go to the far right (it's the only way to make it larger) ? This create a misalignement !

**Comment by mrleblanc101** [comment]
@crynobone 
Changing this
```
      :class="{
        'md:w-4/5': fullWidthContent,
        'md:w-3/5': !fullWidthContent,
        'w-full md:py-5': !field.stacked,
        'w-full md:pt-2': field.stacked,
      }"
```
to this fixed our problem:
```
      :class="{
        'md:w-4/5': fullWidthContent && !field.stacked,
        'md:w-3/5': !fullWidthContent,
        'w-full md:py-5': !field.stacked,
        'w-full md:pt-2': field.stacked,
      }"
```
![MicrosoftTeams-image](https://user-images.githubusercontent.com/17392251/228580974-c5da74b0-a2cc-4071-9d54-2c138f4f7a97.png)

Kinda odd to override a component to change a single line.

**Comment by cjholowatyj** [comment]
I'm in agreement with @mrleblanc101 on this one (I'm in a similar situation). I want to be able to use a Markdown field that makes use of both `stacked()` AND `fullWidth()` at the same time...

Can a use case be thought of for why `stacked()` shouldn't always start out as `w-4/5` and extend to `w-5/5` on `fullWidth()`? I cannot imagine a scenario where I would want `stacked()` to result in a `w-3/5` situation. Either the field runs to `4/5` to match the right margin of all the other fields, or it runs to `5/5` because you need max space. What reason would someone have for needing it `3/5`?

It seems a little like tunnel-vision to make the assumption that the current behaviour is what every developer should expect. If you give developers an additional increment of control over the layout of individual fields, we wouldn't have to rewrite, override, and extend code so much that should already give the developer the freedom to mould it to their use case.
